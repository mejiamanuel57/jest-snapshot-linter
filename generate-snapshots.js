const React = require("react");
const { View, Text, FlatList } = require("react-native");
const TestRenderer = require("react-test-renderer");
const styled = require("styled-components").default;
require("jest-styled-components");

const simpleRenderer = TestRenderer.create(
  <View>
    <Text>Hello World!</Text>
  </View>
);

const HeaderComponent = () => (
  <View>
    <Text>top of the list</Text>
  </View>
);

const FooterComponent = () => (
  <View aLongValue="a-very-long-value-that-is-irrelevant-to-snapshotting">
    <Text>bottom of the list</Text>
  </View>
);

const data = [
  { key: "a" },
  { key: "b" },
  { key: "c" },
  { key: "d" },
  { key: "e" },
  { key: "f" },
  { key: "g" }
];

const ListComponent = ({ header }) => (
  <FlatList
    data={data}
    ListHeaderComponent={header}
    ListFooterComponent={() => <FooterComponent />}
    renderItem={({ item }) => <Text>{item.key}</Text>}
  />
);

const listRenderer = TestRenderer.create(
  <View>
    <ListComponent header={<HeaderComponent />} />
  </View>
);

const renderedObject = TestRenderer.create(
  <View>
    <Text>An error occurred</Text>
    <Text>{JSON.stringify({ prop: "some value" }, null, 2)}</Text>
  </View>
);

const StyledH1 = styled.h1`
  font-size: 1.5em;
  text-align: center;
  color: palevioletred;
`;

const styledObject = TestRenderer.create(<StyledH1 />);

const justAnObject = {
  action: "Viewed",
  attrs: {
    byline:
      "Rosemary Bennett, Education Editor | Nicola Woolcock, Education Correspondent",
    eventTime: "2018-01-01T00:00:00.000Z",
    headline:
      "Caribbean islands devastated by Hurricane Irma, the worst Atlantic storm on record",
    publishedTime: "2015-03-13T18:54:58.000Z",
    topics: [
      {
        name: "Football",
        slug: "football"
      },
      {
        name: "Manchester United FC",
        slug: "manchester-united"
      },
      {
        name: "Chelsea FC",
        slug: "chelsea"
      },
      {
        name: "Arsenal",
        slug: "arsenal"
      },
      {
        name: "Rugby Union",
        slug: "rugby-union"
      }
    ]
  },
  component: "Page",
  object: "Article"
};

const Lengthy = ({ data, long }) => <View data={data} long={long} />;

const LengthTest = props => (
  <View>
    <Lengthy {...props} />
  </View>
);

const lengthTest = TestRenderer.create(
  <LengthTest data={data} long="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa" />
);

const arrayOfComponents = TestRenderer.create([
  <Text key="1">One</Text>,
  <Text key="2">Two</Text>,
  <Text key="3">Three</Text>
]);

const tests = [
  {
    key: "primitives",
    value: [
      {
        boolean: true,
        null: null,
        number: 5,
        string: "test",
        symbol: Symbol("test"),
        undef: undefined
      },
      {
        boolean: true,
        null: null,
        number: 5,
        string: "test",
        symbol: Symbol("test"),
        undef: undefined
      }
    ]
  },
  {
    key: "simple-rn-component",
    value: simpleRenderer.toJSON()
  },
  {
    key: "list-rn-component",
    value: listRenderer.toJSON()
  },
  {
    key: "rendered-object",
    value: renderedObject.toJSON()
  },
  {
    key: "just-an-object",
    value: justAnObject
  },
  {
    key: "a-styled-object",
    value: styledObject.toJSON()
  },
  {
    key: "length-tests",
    value: lengthTest.toJSON()
  },
  {
    key: "array-of-components",
    value: arrayOfComponents.toJSON()
  }
];

tests.forEach(({ key, value }, indx) =>
  test(`${indx}. snapshot for ${key}`, () => expect(value).toMatchSnapshot())
);
