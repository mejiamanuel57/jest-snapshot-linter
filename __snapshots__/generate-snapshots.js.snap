// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`0. snapshot for primitives 1`] = `
Array [
  Object {
    "boolean": true,
    "null": null,
    "number": 5,
    "string": "test",
    "symbol": Symbol(test),
    "undef": undefined,
  },
  Object {
    "boolean": true,
    "null": null,
    "number": 5,
    "string": "test",
    "symbol": Symbol(test),
    "undef": undefined,
  },
]
`;

exports[`1. snapshot for simple-rn-component 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Hello World!
  </Text>
</View>
`;

exports[`2. snapshot for list-rn-component 1`] = `
<View>
  <RCTScrollView
    ListFooterComponent={[Function]}
    ListHeaderComponent={<HeaderComponent />}
    data={
      Array [
        Object {
          "key": "a",
        },
        Object {
          "key": "b",
        },
        Object {
          "key": "c",
        },
        Object {
          "key": "d",
        },
        Object {
          "key": "e",
        },
        Object {
          "key": "f",
        },
        Object {
          "key": "g",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            top of the list
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          a
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          b
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          c
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          d
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          e
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          f
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          g
        </Text>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          aLongValue="a-very-long-value-that-is-irrelevant-to-snapshotting"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            bottom of the list
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`3. snapshot for rendered-object 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    An error occurred
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    {
  "prop": "some value"
}
  </Text>
</View>
`;

exports[`4. snapshot for just-an-object 1`] = `
Object {
  "action": "Viewed",
  "attrs": Object {
    "byline": "Rosemary Bennett, Education Editor | Nicola Woolcock, Education Correspondent",
    "eventTime": "2018-01-01T00:00:00.000Z",
    "headline": "Caribbean islands devastated by Hurricane Irma, the worst Atlantic storm on record",
    "publishedTime": "2015-03-13T18:54:58.000Z",
    "topics": Array [
      Object {
        "name": "Football",
        "slug": "football",
      },
      Object {
        "name": "Manchester United FC",
        "slug": "manchester-united",
      },
      Object {
        "name": "Chelsea FC",
        "slug": "chelsea",
      },
      Object {
        "name": "Arsenal",
        "slug": "arsenal",
      },
      Object {
        "name": "Rugby Union",
        "slug": "rugby-union",
      },
    ],
  },
  "component": "Page",
  "object": "Article",
}
`;

exports[`5. snapshot for a-styled-object 1`] = `
.c0 {
  font-size: 1.5em;
  text-align: center;
  color: palevioletred;
}

<h1
  className="c0"
/>
`;
